### DISCUSSION TITLE
[BUG] only read xxx bytes so close to full but cant read the full what is the problem everyhing read write accesible, enought ram and cpu in well state but repeat hang with no succes.

### DISCUSSION THREAD

**Discussion by spamkutu** [Chia-Network/chia-blockchain#8402] [Support]
**To Reproduce**
its often happen speacialy on network atached storage and other external disk i though at first my ram not enought but iv seen and check that i got i lot of ram empty waiting there. I guess chia lose the connect and new connection not know earlier one beside the some drive.

mostly happen on external driver and on the nas its happen mostly %10 %12 states 

**Expected behavior**
Only read xxxx size but my rams have enought room for that i could also read and write driver there is no problem but this state repeat and cannot fix there must be a way to read full state or if its not recoverable state why that akways repeat repeat and repeat with nothing. I also offering 5 min time for err could be random time 3to7 minutes must better becouse some times all the pilots hangs like this some of them could recoverable but if you not signal random time each try to start some and system hang no one recoverable.

Here by the only read bug logs, just for little we lose all the days ; """ Only read 259199 of 259767 bytes at offset 301726593 """

Starting plotting progress into temporary dirs: N:\TMP and N:\TMP
ID: a40f173c5a843fe0757a28892597c2e09f3bb4d13c4f5cdb009b3cd9683b9c13
Plot size is: 32
Buffer size is: 3890MiB
Using 128 buckets
Final Directory is: N:\10
Using 2 threads of stripe size 65536
Process ID is: 2240

Starting phase 1/4: Forward Propagation into tmp files... Sun Sep  5 19:29:08 2021
Computing table 1
F1 complete, time: 3096.24 seconds. CPU (6.02%) Sun Sep  5 20:20:45 2021
Computing table 2
	Bucket 0 uniform sort. Ram: 3.739GiB, u_sort min: 0.563GiB, qs min: 0.281GiB.
	Bucket 1 uniform sort. Ram: 3.739GiB, u_sort min: 0.563GiB, qs min: 0.281GiB.
	Bucket 2 uniform sort. Ram: 3.739GiB, u_sort min: 0.563GiB, qs min: 0.281GiB.
	Bucket 3 uniform sort. Ram: 3.739GiB, u_sort min: 1.125GiB, qs min: 0.281GiB.
	Bucket 4 uniform sort. Ram: 3.739GiB, u_sort min: 0.563GiB, qs min: 0.281GiB.
	Bucket 5 uniform sort. Ram: 3.739GiB, u_sort min: 0.563GiB, qs min: 0.281GiB.
	Bucket 6 uniform sort. Ram: 3.739GiB, u_sort min: 0.563GiB, qs min: 0.281GiB.
	Bucket 7 uniform sort. Ram: 3.739GiB, u_sort min: 1.125GiB, qs min: 0.281GiB.
	Bucket 8 uniform sort. Ram: 3.739GiB, u_sort min: 0.563GiB, qs min: 0.281GiB.
	Bucket 9 uniform sort. Ram: 3.739GiB, u_sort min: 0.563GiB, qs min: 0.281GiB.
	Bucket 10 uniform sort. Ram: 3.739GiB, u_sort min: 1.125GiB, qs min: 0.281GiB.
	Bucket 11 uniform sort. Ram: 3.739GiB, u_sort min: 1.125GiB, qs min: 0.281GiB.
	Bucket 12 uniform sort. Ram: 3.739GiB, u_sort min: 1.125GiB, qs min: 0.281GiB.
	Bucket 13 uniform sort. Ram: 3.739GiB, u_sort min: 1.125GiB, qs min: 0.281GiB.
	Bucket 14 uniform sort. Ram: 3.739GiB, u_sort min: 0.563GiB, qs min: 0.281GiB.
	Bucket 15 uniform sort. Ram: 3.739GiB, u_sort min: 1.125GiB, qs min: 0.281GiB.
	Bucket 16 uniform sort. Ram: 3.739GiB, u_sort min: 0.563GiB, qs min: 0.281GiB.
	Bucket 17 uniform sort. Ram: 3.739GiB, u_sort min: 1.125GiB, qs min: 0.281GiB.
	Bucket 18 uniform sort. Ram: 3.739GiB, u_sort min: 0.563GiB, qs min: 0.281GiB.
	Bucket 19 uniform sort. Ram: 3.739GiB, u_sort min: 1.125GiB, qs min: 0.281GiB.
	Bucket 20 uniform sort. Ram: 3.739GiB, u_sort min: 0.563GiB, qs min: 0.281GiB.
	Bucket 21 uniform sort. Ram: 3.739GiB, u_sort min: 0.563GiB, qs min: 0.281GiB.
	Bucket 22 uniform sort. Ram: 3.739GiB, u_sort min: 1.125GiB, qs min: 0.281GiB.
	Bucket 23 uniform sort. Ram: 3.739GiB, u_sort min: 0.563GiB, qs min: 0.281GiB.
	Bucket 24 uniform sort. Ram: 3.739GiB, u_sort min: 0.563GiB, qs min: 0.281GiB.
	Bucket 25 uniform sort. Ram: 3.739GiB, u_sort min: 0.563GiB, qs min: 0.281GiB.
	Bucket 26 uniform sort. Ram: 3.739GiB, u_sort min: 0.563GiB, qs min: 0.281GiB.
	Bucket 27 uniform sort. Ram: 3.739GiB, u_sort min: 0.563GiB, qs min: 0.281GiB.
	Bucket 28 uniform sort. Ram: 3.739GiB, u_sort min: 0.563GiB, qs min: 0.281GiB.
	Bucket 29 uniform sort. Ram: 3.739GiB, u_sort min: 1.125GiB, qs min: 0.281GiB.
	Bucket 30 uniform sort. Ram: 3.739GiB, u_sort min: 0.563GiB, qs min: 0.281GiB.
	Bucket 31 uniform sort. Ram: 3.739GiB, u_sort min: 0.563GiB, qs min: 0.281GiB.
	Bucket 32 uniform sort. Ram: 3.739GiB, u_sort min: 1.125GiB, qs min: 0.281GiB.
	Bucket 33 uniform sort. Ram: 3.739GiB, u_sort min: 1.125GiB, qs min: 0.281GiB.
	Bucket 34 uniform sort. Ram: 3.739GiB, u_sort min: 0.563GiB, qs min: 0.281GiB.
	Bucket 35 uniform sort. Ram: 3.739GiB, u_sort min: 1.125GiB, qs min: 0.281GiB.
	Bucket 36 uniform sort. Ram: 3.739GiB, u_sort min: 0.563GiB, qs min: 0.281GiB.
	Bucket 37 uniform sort. Ram: 3.739GiB, u_sort min: 1.125GiB, qs min: 0.281GiB.
	Bucket 38 uniform sort. Ram: 3.739GiB, u_sort min: 0.563GiB, qs min: 0.281GiB.
	Bucket 39 uniform sort. Ram: 3.739GiB, u_sort min: 0.563GiB, qs min: 0.281GiB.
	Bucket 40 uniform sort. Ram: 3.739GiB, u_sort min: 1.125GiB, qs min: 0.281GiB.
	Bucket 41 uniform sort. Ram: 3.739GiB, u_sort min: 0.563GiB, qs min: 0.281GiB.
	Bucket 42 uniform sort. Ram: 3.739GiB, u_sort min: 0.563GiB, qs min: 0.281GiB.
	Bucket 43 uniform sort. Ram: 3.739GiB, u_sort min: 1.125GiB, qs min: 0.281GiB.
	Bucket 44 uniform sort. Ram: 3.739GiB, u_sort min: 0.563GiB, qs min: 0.281GiB.
	Bucket 45 uniform sort. Ram: 3.739GiB, u_sort min: 1.125GiB, qs min: 0.281GiB.
	Bucket 46 uniform sort. Ram: 3.739GiB, u_sort min: 0.563GiB, qs min: 0.281GiB.
Only read 0 of 262143 bytes at offset 0 from "N:\\TMP\\plot-k32-2021-09-05-19-29-a40f173c5a843fe0757a28892597c2e09f3bb4d13c4f5cdb009b3cd9683b9c13.plot.p1.t1.sort_bucket_046.tmp" with length 301986360. Error 1. Retrying in five minutes.
Only read 259199 of 259767 bytes at offset 301726593 from "N:\\TMP\\plot-k32-2021-09-05-19-29-a40f173c5a843fe0757a28892597c2e09f3bb4d13c4f5cdb009b3cd9683b9c13.plot.p1.t1.sort_bucket_046.tmp" with length 301986360. Error 0. Retrying in five minutes.
Only read 259199 of 259767 bytes at offset 301726593 from "N:\\TMP\\plot-k32-2021-09-05-19-29-a40f173c5a843fe0757a28892597c2e09f3bb4d13c4f5cdb009b3cd9683b9c13.plot.p1.t1.sort_bucket_046.tmp" with length 301986360. Error 0. Retrying in five minutes.
Only read 259199 of 259767 bytes at offset 301726593 from "N:\\TMP\\plot-k32-2021-09-05-19-29-a40f173c5a843fe0757a28892597c2e09f3bb4d13c4f5cdb009b3cd9683b9c13.plot.p1.t1.sort_bucket_046.tmp" with length 301986360. Error 0. Retrying in five minutes.
Only read 259199 of 259767 bytes at offset 301726593 from "N:\\TMP\\plot-k32-2021-09-05-19-29-a40f173c5a843fe0757a28892597c2e09f3bb4d13c4f5cdb009b3cd9683b9c13.plot.p1.t1.sort_bucket_046.tmp" with length 301986360. Error 0. Retrying in five minutes.
Only read 259199 of 259767 bytes at offset 301726593 from "N:\\TMP\\plot-k32-2021-09-05-19-29-a40f173c5a843fe0757a28892597c2e09f3bb4d13c4f5cdb009b3cd9683b9c13.plot.p1.t1.sort_bucket_046.tmp" with length 301986360. Error 0. Retrying in five minutes.
Only read 259199 of 259767 bytes at offset 301726593 from "N:\\TMP\\plot-k32-2021-09-05-19-29-a40f173c5a843fe0757a28892597c2e09f3bb4d13c4f5cdb009b3cd9683b9c13.plot.p1.t1.sort_bucket_046.tmp" with length 301986360. Error 0. Retrying in five minutes.
Only read 259199 of 259767 bytes at offset 301726593 from "N:\\TMP\\plot-k32-2021-09-05-19-29-a40f173c5a843fe0757a28892597c2e09f3bb4d13c4f5cdb009b3cd9683b9c13.plot.p1.t1.sort_bucket_046.tmp" with length 301986360. Error 0. Retrying in five minutes.
Could not open N:\TMP\plot-k32-2021-09-05-19-29-a40f173c5a843fe0757a28892597c2e09f3bb4d13c4f5cdb009b3cd9683b9c13.plot.p1.t1.sort_bucket_046.tmp: No such file or directory. Retrying in five minutes.
Only read 259199 of 259767 bytes at offset 301726593 from "N:\\TMP\\plot-k32-2021-09-05-19-29-a40f173c5a843fe0757a28892597c2e09f3bb4d13c4f5cdb009b3cd9683b9c13.plot.p1.t1.sort_bucket_046.tmp" with length 301986360. Error 0. Retrying in five minutes.
Only read 259199 of 259767 bytes at offset 301726593 from "N:\\TMP\\plot-k32-2021-09-05-19-29-a40f173c5a843fe0757a28892597c2e09f3bb4d13c4f5cdb009b3cd9683b9c13.plot.p1.t1.sort_bucket_046.tmp" with length 301986360. Error 0. Retrying in five minutes.
Only read 259199 of 259767 bytes at offset 301726593 from "N:\\TMP\\plot-k32-2021-09-05-19-29-a40f173c5a843fe0757a28892597c2e09f3bb4d13c4f5cdb009b3cd9683b9c13.plot.p1.t1.sort_bucket_046.tmp" with length 301986360. Error 0. Retrying in five minutes.
Only read 259199 of 259767 bytes at offset 301726593 from "N:\\TMP\\plot-k32-2021-09-05-19-29-a40f173c5a843fe0757a28892597c2e09f3bb4d13c4f5cdb009b3cd9683b9c13.plot.p1.t1.sort_bucket_046.tmp" with length 301986360. Error 0. Retrying in five minutes.
Only read 259199 of 259767 bytes at offset 301726593 from "N:\\TMP\\plot-k32-2021-09-05-19-29-a40f173c5a843fe0757a28892597c2e09f3bb4d13c4f5cdb009b3cd9683b9c13.plot.p1.t1.sort_bucket_046.tmp" with length 301986360. Error 0. Retrying in five minutes.
Only read 259199 of 259767 bytes at offset 301726593 from "N:\\TMP\\plot-k32-2021-09-05-19-29-a40f173c5a843fe0757a28892597c2e09f3bb4d13c4f5cdb009b3cd9683b9c13.plot.p1.t1.sort_bucket_046.tmp" with length 301986360. Error 0. Retrying in five minutes.

**Desktop**
- OS: Windows 10
- CPU: Intel(R) Core(TM) i7-10710U

**Comment by loppefaaret** [comment]
i would recommend you switch to plotting on a local drive - it seems like you are potentially dropping the connection to the external drive / network drive, and windows re-establishes the connection - but that resets the filelocks that the plotter is handling, and it looses that lock, and the process fails

**Comment by spamkutu** [comment]
But i have no choice for that i use inuk one ssd gone allready and this is not enought port to work hdds inside as local i could only use usb atached drives and some nat from the lan i have 2 box only two is not have capacity to get inside to my hdds, i will try to do isci but im on windows no server and that support for only connection i will do 3rd parti soliton if you know any well suited iscsi storage server( open source,free or low cost) please share to me i will be happy, and for that case something must do to recall and relock pilot to process not fail doesn't it?

**Comment by loppefaaret** [comment]
We are not working directly on a filesystem layer, we employ the operating system to read and write data - and how that functions is that it put locks on the files as our process is working on the file - if the OS looses the connection, the filelock is lost as well.
are you atleast using a ethernet cable to connect the machine then ? if you are using wireless for networking, that might explain the constant connectivity issues

**Comment by spamkutu** [comment]
>Wednesday, September 8, 2021 4:33 PM +03:00 from loppefaaret ***@***.***>:
> 
> 
>We are not working directly on a filesystem layer, we employ the operating system to read and write data - and how that functions is that it put locks on the files as our process is working on the file - if the OS looses the connection, the filelock is lost as well.
>are you atleast using a ethernet cable to connect the machine then ? if you are using wireless for networking, that might explain the constant connectivity issues
>—
>You are receiving this because you authored the thread.
>Reply to this email directly,  view it on GitHub , or  unsubscribe .
>Triage notifications on the go with GitHub Mobile for  iOS or  Android .  
nope both but mostly eternet side i not use wireless for chia releated if it could correct my problem i beside so slow i could use that too anyway
 
--
spam kutu

**Comment by BrandtH22** [comment]
@spamkutu , since this ticket was opened a number of years ago without further comments it seems this issue has been resolved. 
We will be closing this ticket but if we have closed this ticket in error do not hesitate to reach out to us again with any followup questions or comments.

The best place to reach our support team is on Discord (https://discord.gg/chia) or by reopening this ticket.
